packages/block-library/src/navigation/edit/index.js

```diff
@@ -304,7 +306,13 @@ function Navigation( {
 
 	const isCollapsed =
 		( 'mobile' === overlayMenu && isMobileBreakPoint ) ||
-		'always' === overlayMenu;
+		'always' === overlayMenu ||
+		( 'auto' === overlayMenu && navigationIsWrapping( navRef.current ) );
+
+	// We need to update on window resize so that the nav collapses appropriately.
+	// eslint-disable-next-line no-unused-vars
+	const [ ignored, forceUpdate ] = useReducer( ( x ) => x + 1, 0 );
+	window.addEventListener( 'resize', debounce( forceUpdate, 10 ) );
```

Comment:
I don't like the approach, but I'm struggling to think of anything better. If we go this way we should at least extract it to a hook.